terraform/09-video-hearing-core/20-main.tf

```diff
@@ -258,15 +259,46 @@ module "storage" {
 # VH - SignalR
 #--------------------------------------------------------------
 
+data "azurerm_key_vault" "acmekv" {
+  count               = var.environment != "stg" ? 1 : 0
+  name                = "acmedtssds${var.environment}"
+  resource_group_name = "sds-platform-${var.environment}-rg"
+}
+
+# Staging is using the hearings.reform.hmcts.net wildcard which is only available in the Production KV.
+# Hence requirement for this logic.
+data "azurerm_key_vault" "acmekvstg" {
+  count               = var.environment == "stg" ? 1 : 0
+  name                = "acmedtssdsprod"
+  resource_group_name = "sds-platform-prod-rg"
+  provider            = azurerm.acme_cert
+}
+
 module "SignalR" {
-  source      = "./modules/SignalR"
-  environment = var.environment
+  source = "./modules/SignalR"
 
-  resource_prefix     = "${local.std_prefix}${local.suffix}"
+  name                = "${local.std_prefix}${local.suffix}"
   resource_group_name = azurerm_resource_group.vh-infra-core.name
-  location            = azurerm_resource_group.vh-infra-core.location
+  managed_identities  = [azurerm_user_assigned_identity.vh_mi.id]
+  custom_domain_name  = var.signalr_custom_domain_name
+  key_vault_cert_name = var.environment == "stg" || var.environment == "prod" ? "wildcard-hearings-reform-hmcts-net" : "wildcard-${var.environment}-platform-hmcts-net"
+  key_vault_uri       = var.environment == "stg" ? data.azurerm_key_vault.acmekvstg[0].vault_uri : data.azurerm_key_vault.acmekv[0].vault_uri
+  tags                = local.common_tags
+}
 
-  tags = local.common_tags
+resource "azurerm_role_assignment" "acmmekv_access_policy" {
+  count                = var.environment != "stg" ? 1 : 0
+  role_definition_name = "Key Vault Administrator"
```

Comment:
It should not be Admin access.
please us "Key Vault Secrets User"

Reply:
resolved.